weaviate/datadog_checks/weaviate/check.py

```diff
@@ -0,0 +1,128 @@
+
+                    version_parts = {
+                        'major': major,
+                        'minor': minor,
+                        'patch': patch,
+                    }
+                    self.set_metadata('version', version_raw, scheme='semver', part_map=version_parts)
+                else:
+                    self.log.debug("Invalid Weaviate version format: %s", version)
+            except Exception as e:
+                self.log.debug("Error while parsing Weaviate version: %s", str(e))
+        else:
+            self.log.debug("Could not retrieve version metadata from host.")
+
+    def _submit_liveness_metrics(self):
+        endpoint = urljoin(self.api_url, DEFAULT_LIVENESS_ENDPOINT)
+        tags = self.tags
+        tags.append(f"weaviate_liveness_url:{endpoint}")
+
+        start_time = time.time()
+        response = self.http.get(endpoint)
+        end_time = time.time()
+
+        if response.ok:
+            latency = round_value((end_time - start_time) * 1000, 2)
+            self.service_check('liveness.status', 0, tags)
+            self.gauge('http.latency_ms', latency, tags=tags)
+        else:
+            self.service_check('liveness.status', 2, tags)
+
+    def _submit_node_metrics(self):
+        endpoint = urljoin(self.api_url, DEFAULT_NODE_METRICS_ENDPOINT)
+        response = self.http.get(endpoint)
+        if not response.ok:
+            self.log.debug("Could not retrieve Node metrics. Request returned a: %s", str(response.status_code))
+            return
+        try:
+            data = response.json()
+            # Mapping service checks to HEALTHY = Ok, UNHEALTHY = Warning, UNAVAILABLE = Critical.
+            # Defaults to unknown if not of the above (3).
+            status_values = {'HEALTHY': 0, 'UNHEALTHY': 1, 'UNAVAILABLE': 2}
+
+            for node in data.get('nodes', []):
+                tags = self.tags
+
+                tags.append(f"weaviate_node:{node.get('name')}")
+                tags.append(f"weaviate_version:{node.get('version')}")
+                tags.append(f"weaviate_githash:{node.get('gitHash')}")
+
+                if status := node.get('status'):
+                    tags.append(f"weaviate_node_status:{status.lower()}")
+                    self.gauge('node.status', status_values.get(status, AgentCheck.UNKNOWN), tags=tags)
+                    self.service_check('node.status', status_values.get(status, AgentCheck.UNKNOWN), tags=tags)
+
+                if stats := node.get('stats'):
+                    self.gauge('node.stats.shards', stats.get('shardCount', 0), tags=tags)
+                    self.gauge('node.stats.objects', stats.get('objectCount', 0), tags=tags)
+
+                if shards := node.get('shards'):
+                    for shard in shards:
+                        tags.append(f"weaviate_shard_name:{shard.get('name')}")
+                        tags.append(f"weaviate_shard_class:{shard.get('class')}")
```

Comment:
If the `get` method return None, we will have `weaviate_shard_name:None`. Is it OK or do we prefer default to empty string and have `weaviate_shard_name:` (same question applied to the other tags)